src/DataContainer/MemberContainer.php

```diff
@@ -51,4 +58,35 @@ public function onLoginSaveCallback($value, $dc = null)
 
         return $value;
     }
+
+    /**
+     * alias field save callback.
+     *
+     * @param string             $value
+     * @param DataContainer|null $dc
+     *
+     * @return mixed|string
+     */
+    public function onAliasSaveCallback($value, $dc = null)
+    {
+        $aliasExists = function (string $alias) use ($dc): bool {
+            return $this->connection->executeStatement('SELECT id FROM tl_member WHERE alias=? AND id!=?', [$alias, $dc->id]) > 0;
+        };
+
+        // Generate alias if there is none
+        if (!$value) {
+            $parts = array_filter([$dc->activeRecord->academicTitle, $dc->activeRecord->firstname, $dc->activeRecord->nobilityTitle, $dc->activeRecord->lastname]);
+
+            if (empty($parts)) {
+                $parts = ($dc->activeRecord->username ? [$dc->activeRecord->username] : [$dc->id]);
+            }
+            $value = $this->slug->generate(implode(',', $parts), [], $aliasExists, 'member-');
```

Comment:
why implode with ',' this should be an disallowed character, event thou it will be replaced by Slug it should not be used. Imo '-' is better choice here

Reply:
Good point, I just did it like in the old code, but I think this is good change.